src/Silk.Core/Commands/Tests/TagCommand.cs

```diff
@@ -78,8 +89,168 @@ public async Task Alias(CommandContext ctx, string aliasName, [RemainingText] st
             }
             else
             {
-                await ctx.RespondAsync($"Alias `{aliasName}` that points to tag `{orignalTag}` successfulyl created.");
+                await ctx.RespondAsync($"Alias `{aliasName}` that points to tag `{orignalTag}` successfuly created.");
+            }
+        }
+        
+        [Command]
+        public async Task Create(CommandContext ctx, string tagName, [RemainingText] string content)
+        {
+            Tag? tag = await _tagService.GetTagAsync(tagName, ctx.Guild.Id);
+            if (tag is not null)
+            {
+                await ctx.RespondAsync("Tag with that name already exists!");
+            }
+            if (_reservedWords.Any(tagName.StartsWith))
```

Comment:
What does .StartsWith extension method do? I would expect it to take an argument.

Like, `tagname.StartsWith("content");`

 The name just seems weird.
 
 Or should this just be
 
 `if(_reservedWords.Any(tagName))`

Reply:
`_reservedWords` is a collection of strings, and so in C# you can pass the method group in place of a lambda expression, and the variable gets substituted in. Or at least that's my understanding of it.

It's the same as doing `_reservedWords.Any(w => tagName.StartsWith(w));`

Reply:
yes, but why do you need `.StartsWith`? Doesn't `_reservedWords.Any(tagName)` check the collection of strings to see if the `tagName` is reserved in the collection?

Reply:
I think you're thinking of `.Contains()`, which doesn't exactly fulfill the same purpose. I use `.StartsWith` to prevent tags from interfering with commands. e.g: `!tag create "create a player" To create a player...`, someone that wanted to use that tag could never use it, since one of the tag commands is `!tag create`

Reply:
However, something like `!tag create "how to create a player" ...` would be valid, as the first argument isn't being parsed as a command.

Reply:
Ah, ok, I get it now

Reply:
Small correction to my previous statement regarding lambdas, thanks to @333fred 
![43151662-6AC7-498E-AEA6-1DBD628971BC](https://user-images.githubusercontent.com/42438262/109895849-f3dbc080-7c5d-11eb-9839-d4cc88fee786.jpeg)

Reply:
In C# when you pass in a method name since they are considered first-class citizens, the function gets passed in. The function accepts a string then so you are then able to invoke the function in the Any function to check if the tagName.StartsWith any of the reserved words.